mlir/CMakeLists.txt

```diff
@@ -35,6 +35,14 @@ add_definitions(-DMLIR_CUDA_CONVERSIONS_ENABLED=${MLIR_CUDA_CONVERSIONS_ENABLED}
 # is available
 if ("AMDGPU" IN_LIST LLVM_TARGETS_TO_BUILD)
   set(MLIR_ROCM_CONVERSIONS_ENABLED 1)
+  # Configure ROCm support.
```

Comment:
need to upstream this.

Reply:
This chunk of code is ported from `mlir/tools/mlir-rocm-runner/CMakeLists.txt`. In terms of ROCm dependency, the upstream already have it for `mlir-rocm-runner`, and it is the only dependency that needs it.

If I make the same change to port from `mlir-rocm-runner` to the top level of cmake in the upstream repo, it would look strange as I provided rest of code base access to `ROCM_PATH` variable whereas not consume it anywhere.

Reply:
then in order to minimize changes to MLIR core repository, we need to think should the logic be placed under `mlir/lib/Dialect/MIOpen` in case we don't want to upstream this yet.

Reply:
That's a good idea, will do.